azurecompute/src/main/java/org/jclouds/azurecompute/compute/AzureComputeServiceAdapter.java

```diff
@@ -105,10 +105,10 @@
       final String location = template.getLocation().getId();
       final int[] inboundPorts = template.getOptions().getInboundPorts();
 
-      final String storageAccountName = templateOptions.getStorageAccountName().get();
-      final String virtualNetworkName = templateOptions.getVirtualNetworkName().get();
-      final String reservedIPAddress = templateOptions.getReservedIPName().orNull();
-      final String subnetName = templateOptions.getSubnetName().get();
+      final String storageAccountName = templateOptions.getStorageAccountName();
+      final String virtualNetworkName = templateOptions.getNetworks().isEmpty() ? null : Iterables.getOnlyElement(templateOptions.getNetworks());
```

Comment:
AFAIK VM can be attached to one virtualNetwork only. I agree the end-user should be better notified